File: 1512-number-of-good-pairs/1512-number-of-good-pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        int n= nums.size();
        unordered_map<int, int> ump;
        for(int i=0;i<n;i++) {
            ump[nums[i]]++;
        }
        int ans = 0;
        for(auto i:ump) {
            int a = i.second;
            ans+= a*(a-1)/2;
        }
        return ans;
    }
};

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        ump = {}
        for num in nums:
            if num not in ump:
                ump[num] = 1
            else:
                ump[num] += 1
        ans = 0
        for val in ump.values():
            a = val*(val-1)
            a /=2
            ans += a
        return int(ans)
            
            
        